internal/bminventory/inventory.go

```diff
@@ -3898,7 +3898,13 @@ func (b *bareMetalInventory) changeDNSRecordSets(ctx context.Context, cluster co
 		apiVip := cluster.APIVip
 		ingressVip := cluster.IngressVip
 		if common.IsSingleNodeCluster(&cluster) {
-			apiVip, _ = network.GetMachineCIDRIP(common.GetBootstrapHost(&cluster), &cluster)
+			apiVip, err = network.GetIpForSingleNodeInstallation(&cluster, log)
+			if err != nil {
+				log.WithError(err).Errorf("failed to find ip for DNS record: (%s, %s)",
+					domain.APIINTDomainName, apiVip)
```

Comment:
No need to log apiVip. will be empty...
```suggestion
				log.WithError(err).Errorf("failed to find ip for single node installation")
```